### PR TITLE
Api article list basic functionality

### PR DESCRIPTION
**Pull request by hmvitous**
# [PT Story](https://www.pivotaltracker.com/story/show/171817212)
## Changes proposed in this pull request:
* Adds unit test for Article model, and request spec for Articles list
* Configures FactoryBot with article credentials

## What I have learned working on this feature:
* General knowledge of backend process 
* How to make a second article with factory bot
* Practicing more with binding pry

### CODE REVIEW

**Review by Carltesio — COMMENTED** [review]
Looks ok.

**Comment by ghost** [comment]
OK

File: config/routes.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1 do
+      resources :articles, only:[:index, :show]
```
Comment (janko101) [inline config/routes.rb:4]: Show action is next I think, so for now we only need index here. But, I guess it doesn't hurt ...

File: db/migrate/20200321094716_create_articles.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateArticles < ActiveRecord::Migration[6.0]
+  def change
+    create_table :articles do |t|
+      t.string :title
+      t.text :content
+      t.text :snippet
```
Comment (janko101) [inline db/migrate/20200321094716_create_articles.rb:6]: Just make sure you're sending in order: title, snippet, content

File: spec/requests/api/v1/articles_are_stored_in_index_list_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    it 'checks if snippet is correct' do
+      expect(JSON.parse(response.body)["articles"].first["snippet"]).to eq "Is space the next place?"
+    end
+
+    it 'checks if content is correct' do
+      expect(JSON.parse(response.body)["articles"].first["content"]).to eq "Lau has become the president of space."
+    end
+  
+  end
+end
```
Comment (janko101) [inline spec/requests/api/v1/articles_are_stored_in_index_list_spec.rb:33]: Just to remove the space

File: spec/requests/api/v1/articles_are_stored_in_index_list_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    end
+
+    it 'returns 200 status' do
+      expect(response.status).to eq 200
+    end
+    
+    it 'returns correct number of articles' do
+      binding.pry
+      expect(JSON.parse(response.body)["articles"].count).to eq 2
+    end
```
Comment (janko101) [inline spec/requests/api/v1/articles_are_stored_in_index_list_spec.rb:18]: Remove a binding.pry

**Comment by tochman** [comment]
This is NOT okay.

**Pull request merged**